Sources/AEPOptimize/Optimize.swift

```diff
@@ -502,6 +507,45 @@ public class Optimize: NSObject, Extension {
         dispatch(event: event)
     }
 
+    /// Processes debug events triggered by the system.
+    ///
+    /// A debug event allows the optimize extension to processes non-production workflows.
+    /// - Parameter event: the debug `Event` to be handled.
+    private func processDebugEvent(_ event: Event) {
+        guard event.isDebugEvent else {
+            Log.debug(label: OptimizeConstants.LOG_TAG,
+                      " Ignoring Debug event, either handle type is not com.adobe.eventType.system or source is not com.adobe.eventSource.debug")
+            return
+        }
```

Comment:
would it be better here to validate that our `debug.eventType` and `debug.eventSource` values exist and are something we want to handle?

Reply:
Currently I am not receiving debug object in event's data.
Here is an example -
{
  "id": "D2C4E6D7-EF0A-406F-8460-DBD179EFA7D3",
  "name": "AEP Response Event Handle (Spoof)",
  "type": "com.adobe.eventType.system",
  "source": "com.adobe.eventSource.debug",
  "data": {
    "type": "personalization:decisions",
    "payload": [
      {
        "scopeDetails": {
          "decisionProvider": "TGT",
          "activity": {
            "id": "1451350"
          },
          "experience": {
            "id": "1"
          },
          "correlationID": "1451350:1:0",
          "strategies": [
            {
              "step": "entry",
              "trafficType": "0"
            },
            {
              "trafficType": "0",
              "step": "display"
            },
            {
              "trafficType": "0",
              "step": "custom-metric-3"
            },
            {
              "step": "conversion",
              "trafficType": "0"
            }
          ],
          "characteristics": {
            "eventToken": "ozJDe5qxmh1ynJiKM3qS7w6taum/Ule9G8NTrZHPdMm7I2aV5okLvdambMy5fCYKvxeaHLn+wzXtAu/7/KZ4kA=="
          }
        },
        "id": "AT:eyJhY3Rpdml0eUlkIjoiMTQ1MTM1MCIsImV4cGVyaWVuY2VJZCI6IjEifQ==",
        "scope": "optimize-tutorial-loc",
        "items": [
          {
            "data": {
              "content": "<html><body><img src=\"https://www.pngall.com/wp-content/uploads/14/Welcome.png\" style=\"display: block;margin-left: auto;margin-right: auto;width: 80%;\"/></body></html>",
              "id": "0",
              "format": "text/html"
            },
            "id": "0",
            "schema": "https://ns.adobe.com/personalization/html-content-item",
            "meta": {
              "profile.favoriteColor": "black",
              "profile.Happy_PS": "[hello Happy Profile Script, , , optimize-tutorial-loc, , landing url is, ]",
              "profile.daysSinceLastVisit": "5.0",
              "activity.name": "Optimize Tutorial Activity",
              "profile.marketingCloudVisitorId": "12797319901314597948242657659519290950",
              "profile.impressions": "40",
              "experience.name": "Experience A",
              "profile.greetingsAffinity": "[Good afternoon, Good evening]",
              "profile.tntId": "12797319901314597948242657659519290950-lKYpJH"
            }
          }
        ]
      }
    ]
  },
  "timestamp": "2024-11-19 03:13:26 +0000",
  "responseId": "nil",
  "parentId": "nil",
  "mask": "nil"
}